### DISCUSSION TITLE
Error when providing an unknown option

### DISCUSSION THREAD

**Discussion by roberthoglund** [spectreconsole/spectre.console#1339] [General]
First I would like to apologise if what I'm asking for is in the documentation but if so, I couldn't conjure up the correct search phrase.

I would like to abort a command if it receives an option it doesn't recognise. My use case is that I have an option named --dry-run for many commands, which if supplied the command doesn't do any updates. It's a safety thing so if I were to misspell it, e.g. --dyr-run, the command would run and do the update while I'm assuming it won't.

I'm hoping either of these suggestions are possible.

1. There is a setting to set this.
2. There is currently not a setting for this, but it can be added if someone did the work (Which I would volunteer to do)
3. There is no setting, and it's not a suitable feature. But here is how you can do it your self.

**Comment by nils-a** [comment]
You forgot one point in your enumeration of cases:
There is no such setting, **but** the required information is already there.

When your Command [Executes](https://spectreconsole.net/api/spectre.console.cli/icommand/0a6c71a9), it gets a reference to a [`CommandContext`](https://spectreconsole.net/api/spectre.console.cli/commandcontext/):
This context has a property `Remaining` of type [`IRemainingArguments`](https://spectreconsole.net/api/spectre.console.cli/iremainingarguments/) which will contain all arguments that were not used to populate the command/branches and corresponding settings.

So, for your case if `context.Remaining.Parsed.Any() || context.Remaining.Raw.Any()` is what you are looking for, I guess.

**Comment by roberthoglund** [comment]
Perfect, that's just what I need.
